src/evaluate.jl

```diff
@@ -194,26 +194,33 @@ it's evaluated at zero. Note that the syntax `a(vals)` is equivalent to
 `evaluate(a, vals)`; and `a()` is equivalent to `evaluate(a)`.
 """
 function evaluate(a::HomogeneousPolynomial{T}, vals::NTuple{N,S} ) where
-        {T<:Number, S<:NumberNotSeriesN, N}
+        {T<:Number, S<:Number, N}
 
     @assert N == get_numvars()
 
+    return _evaluate(a, vals)
+end
+
+function _evaluate(a::HomogeneousPolynomial{T}, vals::NTuple{N,S} ) where
+        {T<:Number, S<:Number, N}
+
     ct = coeff_table[a.order+1]
     R = promote_type(T,S)
     suma = zero(R)
 
     for (i,a_coeff) in enumerate(a.coeffs)
         iszero(a_coeff) && continue
-        tmp = vals[1]^(ct[i][1])
+        @inbounds tmp = vals[1]^(ct[i][1])
         for n in 2:N
-            tmp *= vals[n]^(ct[i][n])
+            @inbounds tmp *= vals[n]^(ct[i][n])
         end
         suma += a_coeff * tmp
```

Comment:
I think you need `abs(a_coeff)` here?

Reply:
Why? The actual coefficient is `a_coeff`...

Reply:
Yes, sorry, you're right -- we're just trying to bound the polynomial.